internal/xds/rbac/matchers.go

```diff
@@ -183,7 +183,7 @@ func matchersFromPrincipals(principals []*v3rbacpb.Principal) ([]matcher, error)
 			// The source ip principal identifier is deprecated. Thus, a
 			// principal typed as a source ip in the identifier will be a no-op.
 			// The config should use DirectRemoteIp instead.
-		case *v3rbacpb.Principal_RemoteIp:
+		case *v3rbacpb.Principal_RemoteIp: // Allow equating RBAC's direct_remote_ip...do we need this?
```

Comment:
> Allow equating RBAC's direct_remote_ip...do we need this?

Who is this question addressed to?

> New validation should occur for HttpConnectionManager to allow equating RBAC's direct_remote_ip and remote_ip

What does this mean? Should these fields always have the same value? Or what validation are you supposed to do?

Reply:
I think it essentially means that when given those two policies in the configuration for RBAC, to logically treat them as the same thing, which is why the two fields in HCM have the validations they do, in order to support this concept of being the same thing logically. If you break one of these validations for the HCM, DirectRemoteIP != RemoteIP.

Reply:
I think it would be best to add that functionality in a later PR along with a few small changes at the RBAC layer.

Reply:
How about something to the effect of:

```
// gRPC does not distinguish between `direct_remote_ip` and `remote_ip` fields.
// We have validations to ensure that these two fields equate to the same thing. 
// Hence ignoring this field here.
```